semver-gen.sh

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env bash
+#
+# This script generates semantic versioning tags for auto-tagging.
+# It uses the last manual tag and counts the commits since then,
+# as well as adding current commit hash.
+#
+# Syntax looks rather complex as we must also rely on very old git versions.
+#
+
+# Be picky
+set -e
+set -u
+# But not too much
+set +o pipefail
+
+
+############################################################
+# Global variables
+############################################################
+
+# Default tag if none exists yet
+DEFAULT_TAG="v0.1.0"
+
+# Git commit hash length
+DEFAULT_HASH_LENGTH="7"
+
+# Regex to find last manual tag. E.g.: v0.1.2
+TAG_REGEX="v[0-9]+\.[0-9]+\.[0-9]+"
+
+
+############################################################
+# Functions
+############################################################
+
+# Get the commit line with the last manual tag (specified by defined TAG_REGEX).
+# If no such tag exists, return the first commit ever created.
+get_commit_line() {
+	local tag_regex="${1}"
+	local line=
+
+	# Note: git log --decorate=full produces a line similar to this one:
+	# commit 4e6c82beff6ab4175f4d378b6442c8269a87dd8f (tag: refs/tags/v1.0.0, refs/remotes/origin/master, refs/heads/master)
+
+	# Commit line with last manual tag
+	line="$( git log --decorate=full 2>/dev/null \
+		| grep ^commit \
+		| grep -E "tags/${tag_regex}(\)|[[:space:]]|,)" \
+		| head -1 )"
+
+	# First commit line ever created in git
+	if [ -z "${line}" ]; then
+		line="$( git log 2>/dev/null | grep -E '^commit[[:space:]]+[a-f0-9]+' | tail -1 )"
+	fi
+
+	echo "${line}"
+}
+
+# Extract the git tag from the provided git commit line.
+# If no tag exists in that line, use the default tag.
+extract_git_tag() {
```

Comment:
This function does not work as expected in situation if there is no tags set in repository.

Reply:
Then what is the expected behaviour?